components/constellation/constellation.rs

```diff
@@ -1244,13 +1239,17 @@ where
                     warn!("Sending reply to get browsing context failed ({:?}).", e);
                 }
             },
+            // TODO(mandreyel): maybe change semantics of this message to
```

Comment:
I think I'll leave this to another PR (I'd like to do it!) to keep things neatly isolated, if that's alright.

Reply:
Hmm, yeah, this could actually be coalesced into a single message to constellation. Something like `ScriptMsg::GetBrowsingContextInfo(PipelineId)` that returns the most relevant fields (in this case `browsing_context_id` and `parent_pipeline_id`).